src/linear_algebra/tensor_product.lean

```diff
@@ -356,14 +356,23 @@ theorem smul.aux_of {R' : Type*} [has_scalar R' M] (r : R') (m : M) (n : N) :
   smul.aux r (free_add_monoid.of (m, n)) = (r • m) ⊗ₜ[R] n :=
 rfl
 
-variables [smul_comm_class R R' M] [smul_comm_class R R' N]
-variables [smul_comm_class R R'' M] [smul_comm_class R R'' N]
+variables [smul_comm_class R R' M]
+variables [smul_comm_class R R'' M]
 
--- Most of the time we want the instance below this one, which is easier for typeclass resolution
--- to find. The `unused_arguments` is from one of the two comm_classes - while we only make use
--- of one, it makes sense to make the API symmetric.
-@[nolint unused_arguments]
-instance has_scalar' : has_scalar R' (M ⊗[R] N) :=
+/-- Given two modules over a commutative semiring `R`, if one of the factors carries a
+(distributive) action of a second type of scalars `R'`, which commutes with the action of `R`, then
+the tensor product (over `R`) carries an action of `R'`.
+
+This instance defines this `R'` action in the case that it is the left module which has the `R'`
+action. Two natural ways in which this situation arises are:
+ * Extension of scalars
+ * A tensor product of a group representation with a module not carrying an action
+
+Note that in the special case that `R = R'`, since `R` is commutative, we just get the usual scalar
+action on a tensor product of two modules. This special case is important enough that, for
+perfomance reasons, we explicitly define it below, and reduce the priority of this instance. -/
+@[priority 500]
```

Comment:
It's not clear to me that there's any point reducing the priority here; we didn't need it before.

Reply:
Agreed. I actually just mindlessly copied this priority from the instance I deleted in the other file (`ring_theory/tensor_product.lean`) but when you put me on the spot I confess I cannot justify my claim. I'll remove these priorities.